src/controllers/swapAndBridge/swapAndBridge.ts

```diff
@@ -745,8 +747,57 @@ export class SwapAndBridgeController extends EventEmitter {
         ) {
           let routeToSelect
           let routeToSelectSteps
+          let routes = quoteResult.routes || []
+
+          // Exclude routes that require an additional protocol fee in a specific token
+          // if the user does not have sufficient balance of that token.
+          try {
+            routes = routes.filter((route) => {
+              let shouldFilterOut = false
+              if (!route.userTxs) return !shouldFilterOut
+
+              const bridgeTx = route.userTxs.find((tx) => tx.userTxType === 'fund-movr') as
+                | SocketAPIBridgeUserTx
+                | undefined
+
+              if (!bridgeTx) return false
+
+              const bridgeStep = bridgeTx.steps.find((s) => s.type === 'bridge') as
+                | SocketApiBridgeStep
+                | undefined
+
+              if (!bridgeStep) return false
+
+              const normalizedProtocolFeeToken = normalizeIncomingSocketToken(
+                bridgeStep.protocolFees.asset
+              )
+              const protocolFeeTokenNetworkId = this.#networks.networks.find(
+                (n) => Number(n.chainId) === normalizedProtocolFeeToken.chainId
+              )?.id!
+              const tokenToPayFeeWith = this.portfolioTokenList.find(
+                (t) =>
+                  t.address === normalizedProtocolFeeToken.address &&
+                  t.networkId === protocolFeeTokenNetworkId &&
+                  Number(getTokenAmount(t) >= Number(bridgeStep.protocolFees.amount))
+              )
+
+              // if the required protocolFee token is not in the users portfolio
+              // filter out that route to prevent errors on the signAccountOp screen due to
+              // failing simulation and estimation because the user doesn't have sufficient balance to pay the fee
+              if (!tokenToPayFeeWith) shouldFilterOut = true
+
+              return !shouldFilterOut
+            })
+          } catch (error) {
+            console.error(error)
```

Comment:
Should continue further if the logic breaks, right? Because it will behave just like it was before these changes. I tend to agree.

Reply:
yes, the idea is to use a try-catch block to handle cases where certain properties in the nested structure might be missing, possibly due to a socket mistake. If the code inside the block throws, the quote will default to the original routes array